pkg/kubelet/kubelet.go

```diff
@@ -1088,6 +1089,7 @@ func (kl *Kubelet) syncPod(pod *api.BoundPod, dockerContainers dockertools.Docke
 				}
 				glog.Infof("pod %q container %q is unhealthy. Container will be killed and re-created.", podFullName, container.Name, live)
 			} else {
+				podChanged = true
```

Comment:
The only reason killing infra pod container when a container is changed is that we don't know if network related config is changed, such as port.